### DISCUSSION TITLE
Does yabai solve the lag after switching spaces

### DISCUSSION THREAD

**Discussion by Allaman** [asmvik/yabai#1256] [General]
Hello,

I think about installing yabai because of [this]https://piunikaweb.com/2021/12/14/macos-12-monterey-workspace-switching-animation-lag-with-promotion/) issue with the (input) lag after switching spaces.  My main concerns with yabai are, besides disabling SIP, ...
-  if yabai can solve this problem and the focus switching is instant (as I was used to on Linux)
-  if I can disable the tiling feature for all apps. I played around with tiling WMs on Linux (i3, awesome) a couple of times but they do not match my workflow.

I also asked the question on [Reddit](https://www.reddit.com/r/MacOS/comments/ufxjry/yabai_and_focusing_spaces/) with some more details.

Any feedback is appreciated! 

Thanks

**Comment by tg0h** [comment]
hello!

- yes, if you disable SIP, switching spaces happens instantly
- to disable the auto tiling feature (i'm assuming you're referring to the binary space partition feature where new windows are automatically placed on the screen), you will need to tell yabai not to manage the window
```
yabai -m rule --add app='^System Information$' manage=off 
```

(btw, I used your dotfiles repo a few months back to finally set up my neovim for development, many thanks! 🙏  :) )

**Reply by peppy** [reply]
To clarify, if you want to disable BSP for *all* windows (I think that's what the OP was after) you need to change the mode on the space itself:

`yabai -m space 4 --layout float`

where `4` is the space you want to set to "floating" ie. unmanaged.

**Reply by asmvik** [reply]
To disable tiling globally, just use `yabai -m config layout float`

**Reply by Allaman** [reply]
Thanks for your hints. Meanwhile I disabled SIP and configured yabai and skhd. Everything works flawless and snappy. 

- [yabairc](https://github.com/Allaman/dotfiles/blob/master/yabairc)
- [skhdrc](https://github.com/Allaman/dotfiles/blob/master/skhdrc)

> btw, I used your dotfiles repo a few months back to finally set up my neovim for development, many thanks!

Glad to hear!

**Comment by nahuel** [comment]
@tg0h if you disable SIP then switching spaces happens instantly, without having yabai installed?

**Reply by peppy** [reply]
No, you need to have yabai loaded for this to be the case.

**Comment by mil-ad** [comment]
I have SIP disabled and there's still a significant delay when switching workspaces/desktops through yabai/skhd.

It does however seem to be app-dependent. It usually happens when firefox is in the source/destination desktop.

**Reply by Allaman** [reply]
My main browser is Firefox, so at least one instance is always running. No issues, respectively significant improvement with yabai/skhd in comparison to before.

**Reply by mil-ad** [reply]
are you running HEAD or the latest release?

**Reply by Allaman** [reply]
yabai-v5.0.2

**Reply by Allaman** [reply]
Looks like I spoke too soon. :( Suddenly, I experienced a notable lag even after a reboot and with no application opened...

**Reply by mil-ad** [reply]
😢

**Reply by Allaman** [reply]
I figured it out why, maybe it is the case for you as well...!

I just recently added `command -v gopass >/dev/null 2>&1 && export TOKEN=$(gopass show ....)` to my `.zshenv`. Looks like this is evaluated every time I invoke `yabai -m  space focus next/prev` causing the lag. After removing it, I got my snappiness back.

TIL: Keep your shell lag free to keep yabai lag free 😆

**Reply by aaw5017** [reply]
Thanks so much for this info! I just installed yabai and skhd on a new Apple M2 machine. Noticed lag while switching spaces, and after switching all of my logic from `.zshenv` to `.zshrc`, the lag went away completely. Awesome find

**Comment by mil-ad** [comment]
Interesting! I just tried adding a `sleep 10` to my `.zshrc` and it doesn't seem to affect workspace switching.

**Reply by Allaman** [reply]
Yeah, but I added it to my `.zshenv`. Just confirmed it by adding `sleep 5` to it ...

**Comment by tgienger** [comment]
Is there a workaround that will allow disabling SIP and still be able to run ios apps?

**Comment by psygo** [comment]
How was your experience with disabling SIP so far? Have you guys faced any malware threats yet?

I feel so annoyed by this lag in switching spaces, but disabling protection seems dodgy to me. Kinda bonkers that Apple made switching spaces so tied to security.

**Reply by mil-ad** [reply]
I have SIP disabled space switching is fine. But do not expect i3 in Linux speeds, it's still slow.

**Reply by psygo** [reply]
That's so frustrating to hear. I really love i3 and Linux WTM.

**Reply by Allaman** [reply]
I had no issues with SIP disabled, and the switching is now useable. Definitely, it is not as snappy as my i3 ThinkPad, but I guess you can't really compare a minimal window manager with macOS ;)

Maybe https://github.com/nikitabobko/AeroSpace is for you. I don't know if it addresses the lag. I don't plan to switch because I don't like that they are not using the macOS default workspace system.

**Comment by psygo** [comment]
I'm on MacOS Sonoma 14.3 and yabai v6.0.6, trying to configure this thing after disabling SIP, and following the [configure scripting addition documentation](https://github.com/koekeishiya/yabai/wiki/Installing-yabai-(latest-release)#configure-scripting-addition), but it seems to not be working.

```sh
$ sudo yabai --load-sa
sudo: sorry, you are not allowed to set the following environment variables: TERMINFO
```

Does anyone know what's going on?

**Reply by tg0h** [reply]
I get the same warning on sonoma 14.2, yabai 6.0.6. i disabled SIP. yabai works for me...

**Reply by psygo** [reply]
Yabai still works for me. But, even with the `--load-sa` message, is it working? I mean, do you get no lag?

**Reply by tg0h** [reply]
yups, i don't get any lag - switching spaces is instantaneous

**Comment by spaceunifyfifty** [comment]
One cause of slow yabai commands is actually skhd being slowed down by fish shell. I solved it here https://github.com/koekeishiya/yabai/issues/2448
